### DISCUSSION TITLE
Q) Is 7800 supported with cc65?

### DISCUSSION THREAD

**Discussion by zezba9000** [cc65/cc65#1924] [Q&A]
As the title says. Does the cc65 compiler support the Atari 7800 features or only 2600 features?

Or am I asking the wrong question with my limited knowledge of the hardware so far?

**Comment by Fabrizio-Caruso** [comment]
The Atari 5200, Atari 2600 consoles and Atari 8-bit computers are supported.
I wish Atari 7800 were supported.
You can use the none target to create binaries for the Atari 7800 but you have to do more work.

**Comment by Fabrizio-Caruso** [comment]
This is the only example, I know of, of an attempt to use CC65 to code a game for the Atari 7800:
https://sourceforge.net/projects/rfk7800/files/rfk7800/

**Comment by zezba9000** [comment]
Ok tnx. Backed it up for ref.

**Comment by zezba9000** [comment]
Its odd there isn't more 7800 and Jaguar support for C. Both those console can just do more and have a lot more ram to work with.

**Comment by Fabrizio-Caruso** [comment]
It depends on having a volunteer that implements it. No one has decided not to have support for it. The same is true for the BBC Micro, which lacks support despite being a popular system. CC65 can be used on all 6502-based systems to create raw binaries.

**Comment by mrdudz** [comment]
i guess thats because the 7800 is a bit of an obscure platform even in atari world :)

**Comment by polluks** [comment]
@Fabrizio-Caruso There was a BBC target long time ago, see `TGT_BBC`. It needs a cleanup for V2.17 and the info of #579.

**Comment by groessler** [comment]
The 7800 seems to have just 4k of RAM. Very strange, since the older 5200 had 16k. What did the developers smoke? But probably it was forced upon them by upper management.

AFAICT it supports some kind of character mode. The font for this mode would typically have to be stored in the program ROM. Then a light version of conio might be possible.

Who volunteers?

**Comment by karrika** [comment]
I am currently trying to port over Wizzy from Lynx to 7800. Creating a new target was not difficult. There is also a stdjoy driver now that appears to work. The TGI driver is next.

If I get my game ported and working I may have a look at making a pull request.

**Comment by Fabrizio-Caruso** [comment]
@karrika Thanks for the news! Even if you do not port TGI and stdio, but you have a working crt0.s and cfg, could you please share it and/or integrate it in CC65? Some CC65 targets have minimal libs such as for example Supervision (no conio, no stdio, no TGI). Supervision is a usable target (I have written the hello world example in CC65).
Keep up the good work!

**Comment by karrika** [comment]
> @karrika Thanks for the news! Even if you do not port TGI and stdio, but you have a working crt0.s and cfg, could you please share it and/or integrate it in CC65? Some CC65 targets have minimal libs such as for example Supervision (no conio, no stdio, no TGI). Supervision is a usable target (I have written the hello world example in CC65). Keep up the good work!

I don't use version control yet on this port. So sharing is not possible. But if I get the first mission of Wizzy ported then there will be the basic TGI-stuff available with the joystick driver.

Currently I go for the 160 x 242 pixels PAL screen with 4 colours. So no fancy palette tricks. I may also double all display lines to get square pixels. So the real TGI would be atari7800-160-121-4 .

I may use the same font as I have in the Lynx driver.

**Comment by Fabrizio-Caruso** [comment]
@karrika You may be aware of https://sourceforge.net/projects/rfk7800/files/rfk7800/ . It may use more than 4 colors and higher resolution but it is limited to displaying sort of character tiles. It is based on a much older CC65 version, though.

**Comment by karrika** [comment]
Yep. I already compiled that and had a look.

**Comment by Fabrizio-Caruso** [comment]
This discussion is obsolete. @mrdudz the Atari 7800 is a supported target.
